### DISCUSSION TITLE
How to read folder from different repo

### DISCUSSION THREAD

**Discussion by SESA475978** [community/community#22471] [New to GitHub]
<p>Hello there !</p>
<p>I would like to read folders and files from different repository in workflow, how to achieve that.</p>
<p>Currently i am reading it locally as   “solution-file: Backup/${{ github.event.inputs.Date }}/unmanaged/${{ env.TABLES }}.zip”</p>
<p><strong>Current Action reads files from local repo:</strong></p>
<pre><code>- name: Import ${{env.TABLES}} to ${{env.TARGET_ENV_NAME}} Environment
  uses: microsoft/powerplatform-actions/import-solution@v0
  with:
    environment-url: ${{env.TARGET_ENV_URL}}
    app-id: ${{ secrets.APPID }}
    client-secret: ${{ secrets.CLIENT_SECRET }}
    tenant-id: ${{ secrets.TENANT_ID }}
    solution-file: Backup/${{ github.event.inputs.Date }}/unmanaged/${{ env.TABLES }}.zip
    force-overwrite: true
    publish-changes: true
    run-asynchronously: true
</code></pre>
<p>I would like to modify above action to read files from the folder of remote repository.</p>
<p>Thanks in advance !</p>

**Comment by jsoref** [comment, marked as answer]
<aside class="onebox githubblob" data-onebox-src="https://github.com/actions/checkout/blob/main/README.md#Checkout-multiple-repos-nested">
  <header class="source">

      <a href="https://github.com/actions/checkout/blob/main/README.md#Checkout-multiple-repos-nested" target="_blank" rel="noopener">github.com</a>
  </header>

  <article class="onebox-body">
    <h4><a href="https://github.com/actions/checkout/blob/main/README.md#Checkout-multiple-repos-nested" target="_blank" rel="noopener">actions/checkout/blob/main/README.md#Checkout-multiple-repos-nested</a></h4>


      <pre><code class="lang-md">&lt;p align="center"&gt;
  &lt;a href="https://github.com/actions/checkout"&gt;&lt;img alt="GitHub Actions status" src="https://github.com/actions/checkout/workflows/test-local/badge.svg"&gt;&lt;/a&gt;
&lt;/p&gt;

# Checkout V3

This action checks-out your repository under `$GITHUB_WORKSPACE`, so your workflow can access it.

Only a single commit is fetched by default, for the ref/SHA that triggered the workflow. Set `fetch-depth: 0` to fetch all history for all branches and tags. Refer [here](https://help.github.com/en/articles/events-that-trigger-workflows) to learn which commit `$GITHUB_SHA` points to for different events.

The auth token is persisted in the local git config. This enables your scripts to run authenticated git commands. The token is removed during post-job cleanup. Set `persist-credentials: false` to opt-out.

When Git 2.18 or higher is not in your PATH, falls back to the REST API to download the files.

# What's new

- Updated to the node16 runtime by default
  - This requires a minimum [Actions Runner](https://github.com/actions/runner/releases/tag/v2.285.0) version of v2.285.0 to run, which is by default available in GHES 3.4 or later.

# Usage
</code></pre>



  This file has been truncated. <a href="https://github.com/actions/checkout/blob/main/README.md#Checkout-multiple-repos-nested" target="_blank" rel="noopener">show original</a>

  </article>

  <div class="onebox-metadata">
    
    
  </div>

  <div style="clear: both"></div>
</aside>
